### DISCUSSION TITLE
如何获取被@的群成员的信息啊

### DISCUSSION THREAD

**Discussion by wsdd6** [simple-robot/Discussions#176] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

我监听到群消息后 如何获取发消息成员的信息和 消息种被@的成员信息

**Comment by ForteScarlet** [comment]
消息链中（messageContent.messages）会包含若干的 `At` 类型的元素来代表AT消息。
```java
for (var element : messageContent.getMessages()){
   if (element instanceof At at) {
      // 这是一个at消息元素
   }
}
```

得到 `At` 后可以获取到其中被at的人的ID，然后可以通过 `Group.getMember` 或其他类似的API来获取一个目标（比如群成员或者联系人（好友）），进而发送消息。
